To: Nicolas Pitre <nico@cam.org>, Lennert Buytenhek <buytenh@wantstofly.org>
Cc: linux-arm-kernel@lists.arm.linux.org.uk, Sylver Bruneau <sylver.bruneau@googlemail.com>
Subject: [PATCH] Orion: Export the reset button of the QNAP TS-409

The reset button on the QNAP TS-409 is available through gpio.

Signed-off-by: Martin Michlmayr <tbm@cyrius.com>

--- a/arch/arm/mach-orion5x/ts409-setup.c	2008-07-31 04:19:08.000000000 +0000
+++ b/arch/arm/mach-orion5x/ts409-setup.c	2008-07-31 04:21:10.000000000 +0000
@@ -211,10 +211,17 @@
  *     Power button is attached to the PIC microcontroller
  ****************************************************************************/
 
+#define QNAP_TS409_GPIO_KEY_RESET	14
 #define QNAP_TS409_GPIO_KEY_MEDIA	15
 
 static struct gpio_keys_button qnap_ts409_buttons[] = {
 	{
+		.code		= KEY_RESTART,
+		.gpio		= QNAP_TS409_GPIO_KEY_RESET,
+		.desc		= "Reset Button",
+		.active_low	= 1,
+	},
+	{
 		.code		= KEY_COPY,
 		.gpio		= QNAP_TS409_GPIO_KEY_MEDIA,
 		.desc		= "USB Copy Button",
